### PR TITLE
Add MeshShutdownSignaler and update Mesh to shutdown

### PR DESCRIPTION
**Pull request by agunde406**
This commit updates InternalEnvelope to be an Enum with
Message and Shutdown options. Mesh will now has a function
that will return a MeshShutdownSignaler that can be used
to shutdown Mesh. 

A function has also been added to Network to return a
MeshShutdownSignaler as well as Network is just a wrapper
around Mesh.

Signed-off-by: Andrea Gunderson <agunde@bitwise.io>

### CODE REVIEW

File: libsplinter/src/mesh/mod.rs [hunk 1]
```diff
@@ -558,4 +582,65 @@ mod tests {
         let tls = create_test_tls_transport(true);
         test_add_remove_connections(tls, "127.0.0.1:0");
     }
+
+    #[test]
+    // Test mesh can be shutdown
```
Comment (vaporos) [inline libsplinter/src/mesh/mod.rs:587]: Need more descriptive test comments - see PBFT tests for examples.

Comment (agunde406) [reply libsplinter/src/mesh/mod.rs]: updated

File: libsplinter/src/mesh/mod.rs [hunk 2]
```diff
@@ -76,28 +76,10 @@ use crate::mesh::reactor::Reactor;
...
-
-    pub fn payload(&self) -> &[u8] {
-        &self.payload
-    }
-
-    pub fn take_payload(self) -> Vec<u8> {
-        self.payload
-    }
+#[derive(Debug, PartialEq)]
+pub enum InternalEnvelope {
```
Comment (peterschwarz) [inline libsplinter/src/mesh/mod.rs:80]: Should this be pub? Also, with the changes where the signalling is via the `Control` struct, is this still needed to be an enum?

Comment (agunde406) [reply libsplinter/src/mesh/mod.rs]: I kept InternalEnvelope an enum so the recv would still get the shutdown notification. It only needs to be available in with the mesh mod but no where else. There are also a couple other components that are public but do not need to be. I added a commit to reduce their visibility to mesh.

**Pull request merged**